src/taipy/core/_entity/_properties.py

```diff
@@ -18,6 +18,7 @@ class _Properties(UserDict):
     def __init__(self, entity_owner, **kwargs):
         super().__init__(**kwargs)
         self._entity_owner = entity_owner
+        self._previous_data = {}
```

Comment:
I was thinking in the term of data from previous entity but since the purpose is to delay changes in context, `pending_changes` would make much more sense :D